lib/activerecord-import/adapters/postgresql_adapter.rb

```diff
@@ -44,7 +44,7 @@ def post_sql_statements( table_name, options ) # :nodoc:
       if (options[:ignore] || options[:on_duplicate_key_ignore]) && !options[:on_duplicate_key_update] && !options[:recursive]
         sql << sql_for_on_duplicate_key_ignore( table_name, options[:on_duplicate_key_ignore] )
       end
-    elsif options[:on_duplicate_key_ignore] && !options[:on_duplicate_key_update]
+    elsif options[:on_duplicate_key_ignore] && !options[:on_duplicate_key_update] && !options[:on_duplicate_raise]
```

Comment:
I don't think this does what you are thinking. This part of the conditional just prints a warning that the installed version of PostgreSQL doesn't support the option :on_duplicate_key_ignore.